src/logic/transaction.ts

```diff
@@ -127,6 +134,67 @@ export async function checkTransactionData(
   if (!(await contracts.isValidContractAddress(contractAddress))) {
     throw new Error(`Invalid contract address "${contractAddress}"`)
   }
+
+  const minPrice = await config.requireNumber('MIN_SALE_VALUE')
+  const chainName = (await config.requireString('CHAIN_NAME')) as ChainName
+  const salePrice = getSalePrice(params, getMaticChainIdFromNetwork(chainName))
+  if (salePrice !== null && salePrice <= minPrice) {
+    throw new Error(
+      `The transacation data contains a sale price that's lower that the allowed minimum. Sale: ${salePrice}, Min: ${minPrice}`
+    )
+  }
+}
+
+/**
+ * Tries to get the corresponding sale price for the transaction data sent.
+ * It'll return a number (converted from wei) if the data corresponds to any of the sales we're watching and null otherwise
+ * @param params - Transaction data params
+ */
+export function getSalePrice(
```

Comment:
What do you think of moving this to a different utils file like I mentioned above?